Lib/idlelib/debugger.py

```diff
@@ -216,142 +347,131 @@ def interaction(self, message, frame, info=None):
                     m1 = "%s: %s" % (m1, str(value))
                 except:
                     pass
-            bg = "yellow"
         else:
             m1 = ""
             tb = None
-            bg = self.errorbg
-        self.error.configure(text=m1, background=bg)
-        #
-        sv = self.stackviewer
-        if sv:
-            stack, i = self.idb.get_stack(self.frame, tb)
-            sv.load_stack(stack, i)
-        #
-        self.show_variables(1)
-        #
-        if self.vsource.get():
-            self.sync_source_line()
-        #
-        for b in self.buttons:
-            b.configure(state="normal")
-        #
+
+        if m1 != '':
+            self.show_status(m1, error=True)
+        stack, idx = self.idb.get_stack(self.frame, tb)
+        self.clear_stack()
+        for i in range(len(stack)):
+            frame, lineno = stack[i]
+            self.add_stackframe(frame, lineno, current=(i == idx))
+        self.show_vars()
+        self.sync_source_line()
+        self.enable_buttons(self.buttons)
+
         self.top.wakeup()
         # Nested main loop: Tkinter's main loop is not reentrant, so use
         # Tcl's vwait facility, which reenters the event loop until an
         # event handler sets the variable we're waiting on
         self.nesting_level += 1
         self.root.tk.call('vwait', '::idledebugwait')
         self.nesting_level -= 1
-        #
-        for b in self.buttons:
-            b.configure(state="disabled")
-        self.status.configure(text="")
-        self.error.configure(text="", background=self.errorbg)
         self.frame = None
 
+    def show_vars(self):
+        self.vars.move(self.locals, '', 0)
+        self.vars.move(self.globals, '', 1)
+        self.vars.delete(*self.vars.get_children(self.locals))
+        self.vars.delete(*self.vars.get_children(self.globals))
+        self.var_values = {}
+        sel = self.stack.selection()
+        if len(sel) == 1 and sel[0] in self.framevars:
+            locals, globals, _, _ = self.framevars[sel[0]]
+            # note: locals/globals may be from a remotedebugger, in
+            # which case for reasons we don't need to get into here,
+            # they aren't iterable
+            self.add_varheader()
+            for name in sorted(locals.keys(), key=underscore_at_end):
+                self.add_var(name, locals[name])
+            self.add_varheader(isGlobal=True)
+            for name in sorted(globals.keys(), key=underscore_at_end):
+                self.add_var(name, globals[name], isGlobal=True)
+
+    def add_varheader(self, isGlobal=False):
```

Comment:
```suggestion
    def add_varheader(self, is_global=False):
```

Reply:
Please do not add more camelCase names.  We are slowly converting them.  'isglobal' is also possible, but not pretty.  Booleans do not have to have 'is' prefixed.  'global ' by itself is fine as it is an adjective.  'if global' makes perfect sense.
```suggestion
    def add_varheader(self, global=False):
```